src/bsoncxx/test/bson_builder.cpp

```diff
@@ -1702,36 +1699,65 @@ TEST_CASE("list builder with explicit type deduction", "[bsoncxx::builder::list]
         bson_append_int32(&array, "1", -1, 1);
         bson_append_array(&expected, "foo", -1, &array);
 
-        builder::list b{"foo", builder::array{"bar", 1}};
-        bson_eq_object(&expected, b.view().get_document().value);
+        builder::document b{"foo", builder::array{"bar", 1}};
+        bson_eq_object(&expected, b.extract().view());
 
         bson_destroy(&expected);
         bson_destroy(&array);
     }
 
-    SECTION("document") {
-        builder::list b;
-        auto kvp_regex =
-            Catch::Matches("(.*)must be list of key-value pairs(.*)", Catch::CaseSensitive::No);
-        REQUIRE_THROWS_WITH((b = builder::document{"foo", 1, 2}), kvp_regex);
-
-        auto type_regex =
-            Catch::Matches("(.*)must be string type(.*)int32(.*)", Catch::CaseSensitive::No);
-        REQUIRE_THROWS_WITH((b = builder::document{"foo", 1, 2, 4}), type_regex);
-    }
+    // compile-time error
+    // SECTION("document") {
+    //      builder::document b = builder::document{"foo", 1, 2};
+    //      builder::document b = builder::document{"foo", 1, 2, 3};
+    // }
 }
 
-TEST_CASE("empty list builder", "[bsoncxx::builder::list]") {
+TEST_CASE("empty list builder", "[bsoncxx::builder::document]") {
     bson_t expected;
     bson_init(&expected);
 
-    builder::list lst = {};
-    bson_eq_object(&expected, lst.view().get_document().value);
-
     builder::document doc = {};
-    bson_eq_object(&expected, doc.view().get_document().value);
+    bson_eq_object(&expected, doc.extract().view());
 
     builder::array arr = {};
-    bson_eq_object(&expected, arr.view().get_array().value);
+    bson_eq_object(&expected, arr.extract().view());
+}
+
+TEST_CASE("list builder append", "[bsoncxx::builder::document]") {
+    SECTION("document") {
+        bson_t expected;
+        bson_init(&expected);
+
+        bson_append_utf8(&expected, "foo", -1, "bar", -1);
+        bson_append_utf8(&expected, "hello", -1, "world", -1);
+
+        builder::document doc = {"foo", "bar"};
```

Comment:
Would this work with multiple key-pairs?

`builder::list::document doc = { "a", "apple", "b", "banana" };`

Reply:
Hm.  Though it is nice to be able to construct a single key-pair document like `{ "a", 1 }`, it has the potential to confuse users about when they do and don't need to surround their key-value pairs in a `std::pair` with the extra braces.

Since `doc = {{ "a", 1 }}` isn't much worse than `doc = { "a", 1 }`, I'm inclined to say let's not offer that additional convenience constructor for this special case.  Let's make all the ways of using the list builder consistent, even though it makes constructing single key-value documents slightly less nice.  Thoughts?

Reply:
`doc = {{ "a", 1 }}` works as well. For `doc = {"a", 1}`, this is possible because of the converting constructor `document(stdx::string_view key, T&& value);`. Unless that constructor is removed I'm not sure how we could avoid this.